File: src/libqsurfaceformat.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qsurfaceformat_enums = enums;
pub const struct_i32_i32 = struct { first: i32, second: i32 };

/// https://doc.qt.io/qt-6/qsurfaceformat.html
pub const qsurfaceformat = struct {
    /// New constructs a new QSurfaceFormat object.
    ///
    ///
    pub fn New() QtC.QSurfaceFormat {
        return qtc.QSurfaceFormat_new();
    }

    /// New2 constructs a new QSurfaceFormat object.
    ///
    /// ``` options: flag of qsurfaceformat_enums.FormatOption ```
    pub fn New2(options: i64) QtC.QSurfaceFormat {
        return qtc.QSurfaceFormat_new2(@intCast(options));
    }

    /// New3 constructs a new QSurfaceFormat object.
    ///
    /// ``` other: QtC.QSurfaceFormat ```
    pub fn New3(other: ?*anyopaque) QtC.QSurfaceFormat {
        return qtc.QSurfaceFormat_new3(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#operator-eq)
    ///
    /// ``` self: QtC.QSurfaceFormat, other: QtC.QSurfaceFormat ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSurfaceFormat_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setDepthBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetDepthBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetDepthBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#depthBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn DepthBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_DepthBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setStencilBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetStencilBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetStencilBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#stencilBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn StencilBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_StencilBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setRedBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetRedBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetRedBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#redBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn RedBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_RedBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setGreenBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetGreenBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetGreenBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#greenBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn GreenBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_GreenBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setBlueBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetBlueBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetBlueBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#blueBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn BlueBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_BlueBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setAlphaBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat, size: i32 ```
    pub fn SetAlphaBufferSize(self: ?*anyopaque, size: i32) void {
        qtc.QSurfaceFormat_SetAlphaBufferSize(@ptrCast(self), @intCast(size));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#alphaBufferSize)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn AlphaBufferSize(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_AlphaBufferSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setSamples)
    ///
    /// ``` self: QtC.QSurfaceFormat, numSamples: i32 ```
    pub fn SetSamples(self: ?*anyopaque, numSamples: i32) void {
        qtc.QSurfaceFormat_SetSamples(@ptrCast(self), @intCast(numSamples));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#samples)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn Samples(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_Samples(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setSwapBehavior)
    ///
    /// ``` self: QtC.QSurfaceFormat, behavior: qsurfaceformat_enums.SwapBehavior ```
    pub fn SetSwapBehavior(self: ?*anyopaque, behavior: i64) void {
        qtc.QSurfaceFormat_SetSwapBehavior(@ptrCast(self), @intCast(behavior));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#swapBehavior)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    ///
    /// Returns: ``` qsurfaceformat_enums.SwapBehavior ```
    pub fn SwapBehavior(self: ?*anyopaque) i64 {
        return qtc.QSurfaceFormat_SwapBehavior(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#hasAlpha)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn HasAlpha(self: ?*anyopaque) bool {
        return qtc.QSurfaceFormat_HasAlpha(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setProfile)
    ///
    /// ``` self: QtC.QSurfaceFormat, profile: qsurfaceformat_enums.OpenGLContextProfile ```
    pub fn SetProfile(self: ?*anyopaque, profile: i64) void {
        qtc.QSurfaceFormat_SetProfile(@ptrCast(self), @intCast(profile));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#profile)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    ///
    /// Returns: ``` qsurfaceformat_enums.OpenGLContextProfile ```
    pub fn Profile(self: ?*anyopaque) i64 {
        return qtc.QSurfaceFormat_Profile(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setRenderableType)
    ///
    /// ``` self: QtC.QSurfaceFormat, typeVal: qsurfaceformat_enums.RenderableType ```
    pub fn SetRenderableType(self: ?*anyopaque, typeVal: i64) void {
        qtc.QSurfaceFormat_SetRenderableType(@ptrCast(self), @intCast(typeVal));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#renderableType)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    ///
    /// Returns: ``` qsurfaceformat_enums.RenderableType ```
    pub fn RenderableType(self: ?*anyopaque) i64 {
        return qtc.QSurfaceFormat_RenderableType(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setMajorVersion)
    ///
    /// ``` self: QtC.QSurfaceFormat, majorVersion: i32 ```
    pub fn SetMajorVersion(self: ?*anyopaque, majorVersion: i32) void {
        qtc.QSurfaceFormat_SetMajorVersion(@ptrCast(self), @intCast(majorVersion));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#majorVersion)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn MajorVersion(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_MajorVersion(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setMinorVersion)
    ///
    /// ``` self: QtC.QSurfaceFormat, minorVersion: i32 ```
    pub fn SetMinorVersion(self: ?*anyopaque, minorVersion: i32) void {
        qtc.QSurfaceFormat_SetMinorVersion(@ptrCast(self), @intCast(minorVersion));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#minorVersion)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn MinorVersion(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_MinorVersion(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#version)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn Version(self: ?*anyopaque) struct_i32_i32 {
        const _pair: qtc.libqt_pair = qtc.QSurfaceFormat_Version(@ptrCast(self));
        return struct_i32_i32{
            .first = @as(*i32, @ptrCast(@alignCast(_pair.first))).*,
            .second = @as(*i32, @ptrCast(@alignCast(_pair.second))).*,
        };
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setVersion)
    ///
    /// ``` self: QtC.QSurfaceFormat, major: i32, minor: i32 ```
    pub fn SetVersion(self: ?*anyopaque, major: i32, minor: i32) void {
        qtc.QSurfaceFormat_SetVersion(@ptrCast(self), @intCast(major), @intCast(minor));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#stereo)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn Stereo(self: ?*anyopaque) bool {
        return qtc.QSurfaceFormat_Stereo(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setStereo)
    ///
    /// ``` self: QtC.QSurfaceFormat, enable: bool ```
    pub fn SetStereo(self: ?*anyopaque, enable: bool) void {
        qtc.QSurfaceFormat_SetStereo(@ptrCast(self), enable);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setOptions)
    ///
    /// ``` self: QtC.QSurfaceFormat, options: flag of qsurfaceformat_enums.FormatOption ```
    pub fn SetOptions(self: ?*anyopaque, options: i64) void {
        qtc.QSurfaceFormat_SetOptions(@ptrCast(self), @intCast(options));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setOption)
    ///
    /// ``` self: QtC.QSurfaceFormat, option: qsurfaceformat_enums.FormatOption ```
    pub fn SetOption(self: ?*anyopaque, option: i64) void {
        qtc.QSurfaceFormat_SetOption(@ptrCast(self), @intCast(option));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#testOption)
    ///
    /// ``` self: QtC.QSurfaceFormat, option: qsurfaceformat_enums.FormatOption ```
    pub fn TestOption(self: ?*anyopaque, option: i64) bool {
        return qtc.QSurfaceFormat_TestOption(@ptrCast(self), @intCast(option));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#options)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    ///
    /// Returns: ``` flag of qsurfaceformat_enums.FormatOption ```
    pub fn Options(self: ?*anyopaque) i64 {
        return qtc.QSurfaceFormat_Options(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#swapInterval)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn SwapInterval(self: ?*anyopaque) i32 {
        return qtc.QSurfaceFormat_SwapInterval(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setSwapInterval)
    ///
    /// ``` self: QtC.QSurfaceFormat, interval: i32 ```
    pub fn SetSwapInterval(self: ?*anyopaque, interval: i32) void {
        qtc.QSurfaceFormat_SetSwapInterval(@ptrCast(self), @intCast(interval));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#colorSpace)
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn ColorSpace(self: ?*anyopaque) QtC.QColorSpace {
        return qtc.QSurfaceFormat_ColorSpace(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setColorSpace)
    ///
    /// ``` self: QtC.QSurfaceFormat, colorSpace: QtC.QColorSpace ```
    pub fn SetColorSpace(self: ?*anyopaque, colorSpace: ?*anyopaque) void {
        qtc.QSurfaceFormat_SetColorSpace(@ptrCast(self), @ptrCast(colorSpace));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setColorSpace)
    ///
    /// ``` self: QtC.QSurfaceFormat, colorSpace: qsurfaceformat_enums.ColorSpace ```
    pub fn SetColorSpace2(self: ?*anyopaque, colorSpace: i64) void {
        qtc.QSurfaceFormat_SetColorSpace2(@ptrCast(self), @intCast(colorSpace));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setDefaultFormat)
    ///
    /// ``` format: QtC.QSurfaceFormat ```
    pub fn SetDefaultFormat(format: ?*anyopaque) void {
        qtc.QSurfaceFormat_SetDefaultFormat(@ptrCast(format));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#defaultFormat)
    ///
    ///
    pub fn DefaultFormat() QtC.QSurfaceFormat {
        return qtc.QSurfaceFormat_DefaultFormat();
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#setOption)
    ///
    /// ``` self: QtC.QSurfaceFormat, option: qsurfaceformat_enums.FormatOption, on: bool ```
    pub fn SetOption2(self: ?*anyopaque, option: i64, on: bool) void {
        qtc.QSurfaceFormat_SetOption2(@ptrCast(self), @intCast(option), on);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsurfaceformat.html#dtor.QSurfaceFormat)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QSurfaceFormat ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSurfaceFormat_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qsurfaceformat.html#types
pub const enums = struct {
    pub const FormatOption = enum {
        pub const StereoBuffers: i32 = 1;
        pub const DebugContext: i32 = 2;
        pub const DeprecatedFunctions: i32 = 4;
        pub const ResetNotification: i32 = 8;
        pub const ProtectedContent: i32 = 16;
    };

    pub const SwapBehavior = enum {
        pub const DefaultSwapBehavior: i32 = 0;
        pub const SingleBuffer: i32 = 1;
        pub const DoubleBuffer: i32 = 2;
        pub const TripleBuffer: i32 = 3;
    };

    pub const RenderableType = enum {
        pub const DefaultRenderableType: i32 = 0;
        pub const OpenGL: i32 = 1;
        pub const OpenGLES: i32 = 2;
        pub const OpenVG: i32 = 4;
    };

    pub const OpenGLContextProfile = enum {
        pub const NoProfile: i32 = 0;
        pub const CoreProfile: i32 = 1;
        pub const CompatibilityProfile: i32 = 2;
    };

    pub const ColorSpace = enum {
        pub const DefaultColorSpace: i32 = 0;
        pub const SRGBColorSpace: i32 = 1;
    };
};

File: src_original/libqsurfaceformat.cpp
#include <QColorSpace>
#include <QPair>
#include <QSurfaceFormat>
#include <qsurfaceformat.h>
#include "libqsurfaceformat.h"
#include "libqsurfaceformat.hxx"

QSurfaceFormat* QSurfaceFormat_new() {
    return new QSurfaceFormat();
}

QSurfaceFormat* QSurfaceFormat_new2(int options) {
    return new QSurfaceFormat(static_cast<QSurfaceFormat::FormatOptions>(options));
}

QSurfaceFormat* QSurfaceFormat_new3(const QSurfaceFormat* other) {
    return new QSurfaceFormat(*other);
}

void QSurfaceFormat_OperatorAssign(QSurfaceFormat* self, const QSurfaceFormat* other) {
    self->operator=(*other);
}

void QSurfaceFormat_SetDepthBufferSize(QSurfaceFormat* self, int size) {
    self->setDepthBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_DepthBufferSize(const QSurfaceFormat* self) {
    return self->depthBufferSize();
}

void QSurfaceFormat_SetStencilBufferSize(QSurfaceFormat* self, int size) {
    self->setStencilBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_StencilBufferSize(const QSurfaceFormat* self) {
    return self->stencilBufferSize();
}

void QSurfaceFormat_SetRedBufferSize(QSurfaceFormat* self, int size) {
    self->setRedBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_RedBufferSize(const QSurfaceFormat* self) {
    return self->redBufferSize();
}

void QSurfaceFormat_SetGreenBufferSize(QSurfaceFormat* self, int size) {
    self->setGreenBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_GreenBufferSize(const QSurfaceFormat* self) {
    return self->greenBufferSize();
}

void QSurfaceFormat_SetBlueBufferSize(QSurfaceFormat* self, int size) {
    self->setBlueBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_BlueBufferSize(const QSurfaceFormat* self) {
    return self->blueBufferSize();
}

void QSurfaceFormat_SetAlphaBufferSize(QSurfaceFormat* self, int size) {
    self->setAlphaBufferSize(static_cast<int>(size));
}

int QSurfaceFormat_AlphaBufferSize(const QSurfaceFormat* self) {
    return self->alphaBufferSize();
}

void QSurfaceFormat_SetSamples(QSurfaceFormat* self, int numSamples) {
    self->setSamples(static_cast<int>(numSamples));
}

int QSurfaceFormat_Samples(const QSurfaceFormat* self) {
    return self->samples();
}

void QSurfaceFormat_SetSwapBehavior(QSurfaceFormat* self, int behavior) {
    self->setSwapBehavior(static_cast<QSurfaceFormat::SwapBehavior>(behavior));
}

int QSurfaceFormat_SwapBehavior(const QSurfaceFormat* self) {
    return static_cast<int>(self->swapBehavior());
}

bool QSurfaceFormat_HasAlpha(const QSurfaceFormat* self) {
    return self->hasAlpha();
}

void QSurfaceFormat_SetProfile(QSurfaceFormat* self, int profile) {
    self->setProfile(static_cast<QSurfaceFormat::OpenGLContextProfile>(profile));
}

int QSurfaceFormat_Profile(const QSurfaceFormat* self) {
    return static_cast<int>(self->profile());
}

void QSurfaceFormat_SetRenderableType(QSurfaceFormat* self, int typeVal) {
    self->setRenderableType(static_cast<QSurfaceFormat::RenderableType>(typeVal));
}

int QSurfaceFormat_RenderableType(const QSurfaceFormat* self) {
    return static_cast<int>(self->renderableType());
}

void QSurfaceFormat_SetMajorVersion(QSurfaceFormat* self, int majorVersion) {
    self->setMajorVersion(static_cast<int>(majorVersion));
}

int QSurfaceFormat_MajorVersion(const QSurfaceFormat* self) {
    return self->majorVersion();
}

void QSurfaceFormat_SetMinorVersion(QSurfaceFormat* self, int minorVersion) {
    self->setMinorVersion(static_cast<int>(minorVersion));
}

int QSurfaceFormat_MinorVersion(const QSurfaceFormat* self) {
    return self->minorVersion();
}

libqt_pair /* tuple of int and int */ QSurfaceFormat_Version(const QSurfaceFormat* self) {
    QPair<int, int> _ret = self->version();
    // Convert QPair<> from C++ memory to manually-managed C memory
    int* _first = static_cast<int*>(malloc(sizeof(int)));
    int* _second = static_cast<int*>(malloc(sizeof(int)));
    *_first = _ret.first;
    *_second = _ret.second;
    libqt_pair _out;
    _out.first = static_cast<void*>(_first);
    _out.second = static_cast<void*>(_second);
    return _out;
}

void QSurfaceFormat_SetVersion(QSurfaceFormat* self, int major, int minor) {
    self->setVersion(static_cast<int>(major), static_cast<int>(minor));
}

bool QSurfaceFormat_Stereo(const QSurfaceFormat* self) {
    return self->stereo();
}

void QSurfaceFormat_SetStereo(QSurfaceFormat* self, bool enable) {
    self->setStereo(enable);
}

void QSurfaceFormat_SetOptions(QSurfaceFormat* self, int options) {
    self->setOptions(static_cast<QSurfaceFormat::FormatOptions>(options));
}

void QSurfaceFormat_SetOption(QSurfaceFormat* self, int option) {
    self->setOption(static_cast<QSurfaceFormat::FormatOption>(option));
}

bool QSurfaceFormat_TestOption(const QSurfaceFormat* self, int option) {
    return self->testOption(static_cast<QSurfaceFormat::FormatOption>(option));
}

int QSurfaceFormat_Options(const QSurfaceFormat* self) {
    return static_cast<int>(self->options());
}

int QSurfaceFormat_SwapInterval(const QSurfaceFormat* self) {
    return self->swapInterval();
}

void QSurfaceFormat_SetSwapInterval(QSurfaceFormat* self, int interval) {
    self->setSwapInterval(static_cast<int>(interval));
}

QColorSpace* QSurfaceFormat_ColorSpace(const QSurfaceFormat* self) {
    const QColorSpace& _ret = self->colorSpace();
    // Cast returned reference into pointer
    return const_cast<QColorSpace*>(&_ret);
}

void QSurfaceFormat_SetColorSpace(QSurfaceFormat* self, const QColorSpace* colorSpace) {
    self->setColorSpace(*colorSpace);
}

void QSurfaceFormat_SetColorSpace2(QSurfaceFormat* self, int colorSpace) {
    self->setColorSpace(static_cast<QSurfaceFormat::ColorSpace>(colorSpace));
}

void QSurfaceFormat_SetDefaultFormat(const QSurfaceFormat* format) {
    QSurfaceFormat::setDefaultFormat(*format);
}

QSurfaceFormat* QSurfaceFormat_DefaultFormat() {
    return new QSurfaceFormat(QSurfaceFormat::defaultFormat());
}

void QSurfaceFormat_SetOption2(QSurfaceFormat* self, int option, bool on) {
    self->setOption(static_cast<QSurfaceFormat::FormatOption>(option), on);
}

void QSurfaceFormat_Delete(QSurfaceFormat* self) {
    delete self;
}
